### ISSUE TITLE
azure_rm_virtualmachine latest image selection incorrect

### ISSUE THREAD

**Issue by megrehncitrix** [ansible/ansible_collections_azure#601]
##### SUMMARY
In latest azcollection module azure_rm_virtualmachine selects an incorrect latest marketplace image.
The the selection logic assumes that the image list is returned in correct order with latest image as last of the list, this is not the case for some skus and the logic should be updated to retrieve the image flagged as latest, and not rely on an ordered list: https://github.com/ansible-collections/azure/blob/dev/plugins/modules/azure_rm_virtualmachine.py#L1973

Affects sku for generation 1 vm Server 2016:
`"publisher": "MicrosoftWindowsServer", "offer": "WindowsServer", "sku": "2016-Datacenter"`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_virtualmachine

##### ANSIBLE VERSION

```
ansible 2.10.8
  python version = 3.6.8 (default, Nov 16 2020, 16:55:22) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

##### COLLECTION VERSION

```
Collection         Version
------------------ -------
azure.azcollection 1.5.0
```

##### STEPS TO REPRODUCE

Call the module for affected marketplace image: "publisher": "MicrosoftWindowsServer", "offer": "WindowsServer", "sku": "2016-Datacenter"
pass values for below:
```
- name: Create a Windows Virtual Machine
  azure_rm_virtualmachine:
    resource_group: "{{ resource_group }}"
    name: "{{ name }}"
    admin_username: "{{ admin_username }}"
    admin_password: "{{ admin_password }}"
    image:
      offer: WindowsServer
      publisher: MicrosoftWindowsServer
      sku: "2016-Datacenter"
      version: latest

```

##### EXPECTED RESULTS

This should create a vm with the latest image for the sku , which is `14393.4583.2108010852` today.

##### ACTUAL RESULTS
VM is created with the image version of `2016.127.20190416` an old version for sku "2016-Datacenter"

As a temporary workaround, we are pulling the latest marketplace image version from the Azure API before calling the module. Example: `az vm image show --location eastus --urn MicrosoftWindowsServer:WindowsServer:2016-Datacenter:latest`

**Comment by Fred-sun** [comment]
@megrehncitrix I executed your Playbook and the latest image you created should be 2016.127.20190416 for I(SKU = 2016-datacenter). After you log in to the system, the ver of the system is Microsoft Windows [Version 10.0.14393]. Thank you very much!

![image](https://user-images.githubusercontent.com/37327967/129869616-d73857c9-27a2-410e-8ca6-9210ba46f21e.png)

**Comment by megrehncitrix** [comment]
Hi @Fred-sun `Version 10.0.14393` is the Windows version, that's correct. But you'll notice that the image hasn't had updates applied since 2019 on marketplace image version `2016.127.20190416`. The correct latest image is `14393.4583.2108010852` which has had updates applied this month.

**Comment by Fred-sun** [comment]
@megrehncitrix I am very sorry for the trouble caused to you. After reviewing the code logic, I have no way to directly obtain the image of the latest version. What I got was a list of image versions, so I cannot directly determine which version is the latest. So there's no way to fix it yet. I suggest you do that first? Thank you very much!
```e.g.
- name: Create a Windows Virtual Machine
  azure_rm_virtualmachine:
    resource_group: "{{ resource_group }}"
    name: "{{ name }}"
    admin_username: "{{ admin_username }}"
    admin_password: "{{ admin_password }}"
    image:
      offer: WindowsServer
      publisher: MicrosoftWindowsServer
      sku: "2016-Datacenter"
      version: 14393.4583.2108010852

**Comment by megrehncitrix** [comment]
Thanks @Fred-sun It may be as easy as just cleaning up the [get_marketplace_image_version()](https://github.com/ansible-collections/azure/blob/dev/plugins/modules/azure_rm_virtualmachine.py#L1973) function to always pass the provided `version` and return a single value. I'm not able to find the api docs that would confirm the same rest endpoint will return the flagged latest when provided the string `latest` but that seems to be how it works... basically, there's no reason to pull a full list of images and determine the latest if the api does that for you on `latest`.

**Comment by xuzhang3** [comment]
@megrehncitrix #606  will fix this issue.

**Comment by Fred-sun** [comment]
fixes by #606

**Comment by davidgarvey** [comment]
az vm image show --location canadacentral --urn MicrosoftWindowsServer:WindowsServer:2016-Datacenter:latest |python -c 'import json,sys;obj=json.load(sys.stdin);print obj["name"];'
